app/Console/Commands/AdserversCrawlCommand.php

```diff
@@ -0,0 +1,180 @@
+        $this->esc = new Esc(
+            config('app.adshares_wallet'),
+            config('app.adshares_workdir'),
+            config('app.adshares_address'),
+            config('app.adshares_secret'),
+            config('app.adshares_node_host'),
+            config('app.adshares_node_port')
+        );
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->readBroadcasts();
+        $this->crawlHosts();
+    }
+
+    protected function readBroadcasts()
+    {
+        try {
+            $logMessage = $this->esc->getBroadcastLog(time() - $this->registerHostsIfBroadcastedLimit);
+            print_r($logMessage);
+            $logs = $logMessage->broadcast;
+        } catch (\Exception $e) {
+            $logs = [
+                ['message' => bin2hex($this->host), 'address' => config('app.adshares_address'), 'account_msid' => 1],
+            ];
+            var_dump($logs);
+            // currently it enables to process single own host without blockchain
+//             throw $e;
+        }
+
+        // TODO: check with Jacek - smell logic bugs
+        // TODO: review adseelct json generation filters and double check if this could be copied into adserver database
+
+        if ($logs) {
+            foreach ($logs as $log) {
+                $host = hex2bin($log['message']);
+                if ($this->host == $host) {
+                    // found own broadcast. No need to send it again
+                    $this->broadcast = false;
+                }
+                $this->info("Found $host -> {$log['address']}");
+                // TODO: extract algo
+                if (preg_match('/^([a-z0-9][a-z0-9-]{0,62}\.)+([a-z]{2,})$/i', $host)) {
+                    NetworkHost::registerHost(Esc::normalizeAddress($log['address']), $host);
+                // TODO: check this with Jacek in adserver symfony code
+                    // $nHost->setAccountMsid($log['account_msid']);
+                } else {
+                    // TODO: debug error log?
+                }
+            }
+        }
+
+        if ($this->broadcast) {
+            $this->info("Broadcast own host: $this->host");
+            $x = $this->esc->sendBroadcast($this->host);
+            var_dump($x);
+            // TODO: this fails currently
+//             die(print_r($x));
+        }
+    }
+
+    protected function crawlHosts()
+    {
+        $crawlTime = time();
+
+        $hosts = NetworkHost::where('last_seen', '>', time() - $this->crawlHostsIfLastSeenLimit)->get();
+
+        $batch = 0;
+
+        // TODO: integrate with adselect
+        // $adselectService = $this->getContainer()->has('adselect') ? $this->getContainer()->get('adselect') : null;
+        // $adselectService instanceof Adselect;
+        // $adselectCmp = [];
+
+        foreach ($hosts as $r) {
+            $host = $r->host;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "$host" which has the same name as the field declared at line 17. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1117)